gp-includes/routes/translation.php

```diff
@@ -54,10 +61,37 @@ public function import_translations_post( $project_path, $locale_slug, $translat
 			return $this->die_with_404();
 		}
 
-		if ( $this->cannot_and_redirect( 'approve', 'translation-set', $translation_set->id ) ) {
+		$can_import = $this->can( 'import', 'translation-set', $translation_set->id );
+
+		if ( ! $can_import ) {
+			$can_import_waiting = $this->can( 'import-waiting', 'translation-set', $translation_set->id );
+
+			if ( ! $can_import_waiting ) {
+				$this->redirect_with_error( __( 'You are not allowed to do that!', 'glotpress' ), $url );
+				return;
+			}
+		}
+
+		$translation_set_import_status = gp_post( 'status', 'waiting' );
+
+		$allowed_import_stati = array( 'waiting' );
+		if ( $can_import ) {
+			$allowed_import_stati[] = 'current';
+		}
+		if ( ! in_array( $translation_set_import_status, $allowed_import_stati ) ) {
+			$this->redirect_with_error( __( 'Invalid translation status.', 'glotpress' ) );
 			return;
 		}
 
+		add_filter( 'gp_translation_set_import_status', function ( $status ) use ( $translation_set_import_status ) {
+
+			if ( $status !== 'fuzzy' ) {
+				$status = $translation_set_import_status;
+			}
+
+			return $status;
+		});
+
```

Comment:
Instead of adding the filter, why not just pass in the desired status as part of the import call on line 113 below, like:

`
$translations_added = $translation_set->import( $translations, $status );
`